float8_experimental/float8_linear.py

```diff
@@ -71,6 +71,54 @@ def _maybe_initialize_amaxes_scales_for_float8_cast(
         scale.copy_(new_scale)
 
 
+# this code was resurrected from https://github.com/pytorch-labs/float8_experimental/pull/128/files
+@torch._dynamo.allow_in_graph
+class manual_float8_mm(torch.autograd.Function):
+    """
+    Like torch.mm, but with X and W in float8
+    """
+
+    @staticmethod
+    def forward(
+        ctx,
+        x_fp8,
+        w_fp8_t,
+    ):
+        ctx.save_for_backward(x_fp8, w_fp8_t)
+        orig_shape = x_fp8.shape
+        x_fp8_reshaped = x_fp8.reshape(-1, orig_shape[-1])
+        res_bits = torch.mm(x_fp8_reshaped, w_fp8_t)
+        res_bits = res_bits.reshape(*orig_shape[:-1], res_bits.shape[-1])
+        return res_bits
+
+    @staticmethod
+    def backward(ctx, go_fp8):
```

Comment:
nit: align go_fp8 / other naming to the other PR

Reply:
we can do that in separate PRs, since not user facing.  Just keeping things small.

Reply:
I dont know if that changes the size of the PR much but sure thats fine

Reply:
probably just a style difference on how to sequence the renames, either is ok IMO